documentation/src/main/asciidoc/user_guide/query.adoc

```diff
@@ -92,73 +144,468 @@ for (Object book : objectList) {
+</replicated-cache>
+----
+
+[small]*Infinispan storage*
+
+In order to use Infinispan storage with a non-shared index, it's necessary to use LOCAL caches for the indexes:
+
+[source,xml]
+----
+<replicated-cache name="default">
+    <indexing index="ALL">
+        <property name="default.locking_cachename">LuceneIndexesLocking_custom</property>
+        <property name="default.data_cachename">LuceneIndexesData_custom</property>
+        <property name="default.metadata_cachename">LuceneIndexesMetadata_custom</property>
+    </indexing>
+</replicated-cache>
+
+<local-cache name="LuceneIndexesLocking_custom">
+    <indexing index="NONE" />
+</local-cache>
+
+<local-cache name="LuceneIndexesMetadata_custom">
+    <indexing index="NONE" />
+</local-cache>
+
+<local-cache name="LuceneIndexesData_custom">
+    <indexing index="NONE" />
+</local-cache>
+
+----
+
+
+====== near-real-time index manager
+
+Similar to the `directory-based` index manager but takes advantage of the Near-Real-Time features of Lucene. It has better write performance than
+the `directory-based` because it flushes the index to the underlying store less often. The drawback is that it can cause loss of data in case
+of a non-clean shutdown. Can be used in conjunction with `ram`, `filesystem` and local infinispan storage. Configuration for each different storage
+type is the same as the <<query.directory-based>> index manager.
+
+Example with ram:
+
+[source,xml]
+----
+<replicated-cache name="default">
+    <indexing index="ALL">
+        <property name="default.indexmanager">near-real-time</property>
+        <property name="default.directory_provider">ram</property>
+    </indexing>
+</replicated-cache>
+----
+
+Example with filesystem:
+
+[source,xml]
+----
+<replicated-cache name="default">
+    <indexing index="ALL">
+        <property name="default.indexmanager">near-real-time</property>
+    </indexing>
+</replicated-cache>
+----
+
+===== External indexes
+
+Apart from having shared and non-shared indexes managed by Infinispan itself, it is possible to offload indexing to a third party search engine: currently
+Infinispan supports Elasticsearch as a external index storage.
+
+====== Elasticsearch IndexManager (experimental)
+
+This index manager forwards all indexes to an external Elasticsearch server. This is an experimental integration and some features may not be available,
+for example `indexNullAs` is link:https://hibernate.atlassian.net/browse/HSEARCH-2389[not currently supported].
+
+Configuration:
+
+[source,xml]
+----
+<indexing index="LOCAL">
+    <property name="default.indexmanager">elasticsearch</property>
+    <property name="default.elasticsearch.host">http://127.0.0.1:9200</property>
+</indexing>
+
+----
+
+The index mode should be set to `LOCAL`. More information about Elasticsearch integration can be found at the link:https://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#elasticsearch-integration[Hibernate Search manual].
```

Comment:
About the "should be set to `LOCAL`". Should we clarify that it's because the ES cluster can be seen as a "shared index" ?
I'm thinking that we might want to clear this up, as someone might think to run various ES nodes possibly for different Infinispan shards.. yes people are "creative" ;)